File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        tribby = [0, 1, 1]
        for i in range(3, n + 1):
            tribby.append(tribby[-1] + tribby[-2] + tribby[-3])
        return tribby[n]

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        int T[4] = {0, 1, 1, 0};
        for (int i = 3; i <= n; i++, T[0] = T[1], T[1] = T[2], T[2] = T[3])
            T[3] = T[0] + T[1] + T[2];
        return (n < 3) ? T[n] : T[3];   
    }
};